test/e2e/redundancy/route_agent_restart.go

```diff
@@ -48,9 +48,11 @@ func testRouteAgentRestart(f *subFramework.Framework, onGateway bool) {
 	clusterBName := framework.TestContext.ClusterIDs[framework.ClusterB]
 
 	nodes := f.FindNodesByGatewayLabel(framework.ClusterA, onGateway)
+	// This will happen only in "kind" environment because of the flow of
+	// testing gateway failover test
```

Comment:
Not sure i understand this statement. Can you please eloborate, why this will happen only in KIND and not in other deployments? 

Also, gateway failover test even if its executed prior to this, should reset the state of the clusters back to the same state as how it was prior to the test-case, if this is not happening, it should be fixed.
Similarly, IMO these tests can be executed in any order, so this test-case can be executed prior to gateway-failover tests.

Reply:
Hi @sridhargaddam 

As part of the GatewayFailover test refactoring, I'm changing the flow of the test to the following:
One of the clusters will have 2 gateway nodes in active/passive state.
During the failover step, the active gateway node will be crashed and passive gateway become active.

In order to replicate the same test flow in "kind" environment, I'm creating the following infrastructure:
First cluster will have 2 nodes - master and worker.
The worker will serve as GW node.

Second cluster will have 3 nodes - master and 2 worker nodes.
Both worker nodes will serve as GW nodes to replicate the test flow change I described above.

With this infrastructure, the RouteAgentRestart test will not be able to find a non GW node as it request.
This scenatio will never happen in non "kind" environment because there will always be a non GW worker node.

That's why I need this workaround for the "kind" env only.

Hope my explanation made it clear.

Maybe I need to make a note in a test more understandable.

Reply:
@sridhargaddam 

I have updated the comment to make it more clear.